eslint.config.mjs

```diff
@@ -0,0 +1,43 @@
+import frontendEslintConfig from './user-interface/eslint.config.mjs';
+import jsEslintConfig from './common/js-eslint.config.mjs';
+import tsEslintConfig from './common/ts-eslint.config.mjs';
+import jestEslintConfig from './common/jest-eslint.config.mjs';
+
+const frontendSourceConfig = frontendEslintConfig.map((configObject) => ({
+  files: ['user-interface/**/*.ts', 'user-interface/**/*.tsx'],
+  ...configObject,
+}));
+const vitestConfig = frontendEslintConfig.map((configObject) => ({
+  files: ['user-interface/**/*.test.ts', 'user-interface/**/*.test.tsx'],
+  ...configObject,
+}));
+const sourceConfig = tsEslintConfig.map((configObject) => ({
+  files: ['backend/**/*.ts', 'common/**/*.ts', 'dev-tools/**/*.ts', 'test/e2e/**/*.ts'],
+  ...configObject,
+}));
+const jestConfig = jestEslintConfig.map((configObject) => ({
+  files: [
+    'backend/**/*.test.ts',
+    'common/**/*.test.ts',
+    'dev-tools/**/*.test.ts',
+    'test/e2e/**/*.test.ts',
+  ],
+  ...configObject,
+}));
```

Comment:
Treating these—source and tests—differently may not be necessary. I was having a problem that turned out to be related to eslint not being able to determine what version of Jest we are using. As it is, it's possible that if one of the sub-project config files changes, the root would produce a different set of rules for files in that sub-project than the lower level would. I do have an idea for how to test for that though, which honestly we should probably add at some point regardless.

If we don't care to separate source and test rulesets, we would need to modify this and the sub-project configs.